riscv/mmu.h

```diff
@@ -206,14 +206,33 @@ class mmu_t
     return (float128_t){load<uint64_t>(addr), load<uint64_t>(addr + 8)};
   }
 
+  // AMO/Zicbom faults should be reported as store faults
+  // Zicbom instruction has different tval and starting address
+  #define convert_cbo_traps(BODY) \
+    try { \
+      BODY \
+    } catch (trap_load_page_fault& t) { \
+      throw trap_store_page_fault(t.has_gva(), addr, 0, t.get_tinst()); \
+    } catch (trap_store_page_fault& t) { \
+      throw trap_store_page_fault(t.has_gva(), addr, 0, t.get_tinst()); \
+    } catch (trap_load_access_fault& t) { \
+      throw trap_store_access_fault(t.has_gva(), addr, 0, t.get_tinst()); \
+    } catch (trap_store_access_fault& t) { \
+      throw trap_store_access_fault(t.has_gva(), addr, 0, t.get_tinst()); \
+    } catch (trap_load_guest_page_fault& t) { \
+      throw trap_store_guest_page_fault(addr, (addr & ((reg_t(2) << (proc->xlen-1))-1)) >> 2, t.get_tinst()); \
+    } catch (trap_store_guest_page_fault& t) { \
+      throw trap_store_guest_page_fault(addr, (addr & ((reg_t(2) << (proc->xlen-1))-1)) >> 2, t.get_tinst()); \
```

Comment:
I am unfamiliar with the tval2 of the hypervisor. The equation is by source code tracing.

Reply:
Drop the above modification.